test/e2e/suite/cases/probe/probe.go

```diff
@@ -0,0 +1,40 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package probe
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/jetstack/kube-oidc-proxy/test/e2e/framework"
+	"github.com/jetstack/kube-oidc-proxy/test/e2e/framework/helper"
+)
+
+var _ = framework.CasesDescribe("Readiness Probe", func() {
+	f := framework.NewDefaultFramework("readiness-probe")
+
+	It("Should not become ready if the issuer is unavailable", func() {
+		By("Deleting the Issuer so no longer becomes reachable")
+		Expect(f.Helper().DeleteIssuer(f.Namespace.Name)).NotTo(HaveOccurred())
+
+		By("Deleting the current proxy that is ready")
+		Expect(f.Helper().DeleteProxy(f.Namespace.Name)).NotTo(HaveOccurred())
+
+		By("Deploying the Proxy should never become ready as the issuer is unavailable")
+		_, _, err := f.Helper().DeployProxy(f.Namespace, f.IssuerURL(),
+			f.ClientID(), f.IssuerKeyBundle())
+		// Error should occur (not ready)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("Should continue to be ready even if the issuer becomes unavailable", func() {
+		By("Deleting the Issuer so no longer becomes reachable")
+		Expect(f.Helper().DeleteIssuer(f.Namespace.Name)).NotTo(HaveOccurred())
+
+		time.Sleep(time.Second * 10)
```

Comment:
Not a big fan of sleep in testing. Can we use a retry mechanism?

Reply:
In this case specifically we are waiting for the kubelet to hit the readiness probe a couple times. 

I'm not sure how else to test this since it's a no op on the proxy side anyway since it's now always going to return true.

Without waiting "some" amount of time we can't let the kubelet to hit the endpoint a few times

Reply:
What is the problem if we check first that the pod is unready and then ready again?

I think an initial sleep is fine (if we can be 100% sure that the pod has become unready meanwhile), but the success criteria should be retrie

Reply:
The pod should never become unready in this test though.
Here we are just testing that if the issuer becomes unreachable the proxy never becomes unready.

At the start of every test we get a new issuer and proxy that have been tested for readiness.

Reply:
I still don't fully understand why can't recall WaitForPodReady until it succeeds 

But if you really want to keep it like that I am ok with it